site/content/book/v2/helloworldpy.md

```diff
@@ -0,0 +1,170 @@
+## "Hello World" in Python
+This simple "Hello World" example shows how Alice can create an encrypted message to Bob. In that case, we are using [DIDComm Python](https://github.com/sicpa-dlab/didcomm-python) and [Peerdid Python](https://github.com/sicpa-dlab/peer-did-python) libraries from [SICPA](https://www.sicpa.com).
+### Step 1: Install packages
+Once you set up your python environment, you need to install the following packages:
+```
+pip install didcomm
+pip install peerdid
+pip install json
+```
+The code in the following sections can be executed in a single python file or executed interactively in a [Jupyter notebook](https://jupyter.org).
+
+### Step 2: Imports
+First, we need to import all required functions, clases and types from `didcomm` and `peerdid` libraries as follows:
+```
+from didcomm.common.types import DID, VerificationMethodType, VerificationMaterial, VerificationMaterialFormat
+from didcomm.did_doc.did_doc import DIDDoc, VerificationMethod, DIDCommService
+from didcomm.did_doc.did_resolver import DIDResolver
+from didcomm.message import Message
+from didcomm.secrets.secrets_resolver_demo import SecretsResolverDemo
+from didcomm.unpack import unpack, UnpackResult
+from didcomm.common.resolvers import ResolversConfig
+from didcomm.pack_encrypted import pack_encrypted, PackEncryptedConfig, PackEncryptedResult
+from didcomm.secrets.secrets_util import generate_x25519_keys_as_jwk_dict, generate_ed25519_keys_as_jwk_dict, jwk_to_secret
+from peerdid import peer_did
+from peerdid.did_doc import DIDDocPeerDID
+from peerdid.types import VerificationMaterialAuthentication, VerificationMethodTypeAuthentication, VerificationMaterialAgreement, VerificationMethodTypeAgreement, VerificationMaterialFormatPeerDID
+import json
+```
+### Step 3: Helpers
+In this step we add three helper functionalities:
+- **Secret resolver**
+This sample code needs a storage to keep the generated key pair secrets. It will then be referenced by the library as a `secrets_resolver`. We can instantiate it as follows:
+`secrets_resolver = SecretsResolverDemo()`
+Note that the `SecretsResolverDemo` simply stores the keys in a text file named `secrets.json`. As you've just realized, this secret storage is anything but secure. Keep in mind that securing keys is of utmost importance for a self-sovereign identity; never use it in production.
+- **DID Resolver**
+The following class is needed by the library to resolve the DID Peer into a DID Document and put it in the format required by the library:
```

Comment:
This is totally correct, but it may not make sense to people who are coming at this with little context. Also, defining a class right here gets in the way of the overall flow; the reader might wonder why we're suddenly writing classes, when all they wanted to do was create a "Hello World."

My suggestion is to move the class to a separate .py file, not inline in this doc (hyperlink should let them download it) and to tell people to import the file. Perhaps also change your explanatory sentence to a paragraph like this:

"DIDComm only works if your code knows how to resolve DIDs to DID documents. There are various libraries that provide that feature. For example, the [universal resolver](https://github.com/decentralized-identity/universal-resolver) can be used. In this walk-through, we'll provide a simple stub that minimizes dependencies and keeps things as simple as possible. Click [here](peerdidresolver.py) for a file you can import, that will do the trick.

Reply:
I totally agree. Indeed, when I started working on this PR, my dilema was how to make something clear and simple, and at the same time complete. Those resolvers were the main issue, and clearly don't add value and make confusion here.
Based on your idea, I was thinking to replicate the full code with explanations in a Jupiter Notebook. I'll check if I can find a way to have a link that can open it an online kernel like google colab or something similar to kaggle.  
That way, we can keep only the essential code here and all the extra details in the runnable notebook.
I also thinking on adding a new section showing how to initiate a communication. OOB generation, response, and did rotation `from_prior`.

Reply:
new commit pushed including:
- move classes out of the guide
- add links to interactive online Jupiter Notebook. Binder is great! check [this](https://mybinder.org/v2/gh/rodolfomiranda/didcomm-hello-world-py/HEAD?labpath=HelloWorld.ipynb)